datafusion/core/src/optimizer/projection_push_down.rs

```diff
@@ -179,8 +180,12 @@ fn optimize_plan(
                 .map(|f| f.qualified_column())
                 .collect::<HashSet<Column>>();
 
+            let all_column_exprs = new_expr.iter().all(|e| matches!(e, Expr::Column(_)));
```

Comment:
The only question I have here is "what if the column is in an `alias` or something. However since the tests pass 👍